File: src/main/java/com/xuancong/employee_management/model/Salary.java
package com.xuancong.employee_management.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "Salary")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Salary {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private BigDecimal baseSalary;
    private BigDecimal allowance;
    private BigDecimal socialInsurance;
    private BigDecimal healthInsurance;
    private BigDecimal unemploymentInsurance;

    @ManyToOne( fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    private LocalDate endDate;
    private LocalDate effectiveDate;
    private boolean active;
    private String note;


}

File: frontend/models/salary/salary.ts
export type SalaryDetailResponse = {
  id: number;
  employeeId: number;
  employeeName: string;
  branchName: string;
  departmentName: string;
  positionName: string;
  email: string;
  baseSalary: number;               
  allowance: number;               
  socialInsurance: number;          
  healthInsurance: number;      
  unemploymentInsurance: number;    
  effectiveDate: string;            
  active: boolean;
  note: string;
};



export type SalaryCreateRequest = {
  baseSalary: number;
  effectiveDate: string;
  note?: string;
  allowance?: number;
  employeeId: number;
};